src/Servers/HttpSys/src/RequestProcessing/Response.cs

```diff
@@ -32,21 +31,64 @@ internal Response(RequestContext requestContext)
+        {
+            // Reset fields in sequential order according to layout for better memory/cache access behaviour
+            /*
+            Type layout for 'Response'
+            Size: 640 bytes. Paddings: 212 bytes (%33 of empty space)
+            |=================================================================|
+            | Object Header (8 bytes)                                         |
+            |-----------------------------------------------------------------|
+            | Method Table Ptr (8 bytes)                                      |
+            |=================================================================|
+            |   0-7: String _reasonPhrase (8 bytes)                           |
+            |-----------------------------------------------------------------|
+            |  8-15: ResponseBody _nativeStream (8 bytes)                     |
+            |-----------------------------------------------------------------|
+            | 16-23: RequestContext <RequestContext>k__BackingField (8 bytes) |
+            |-----------------------------------------------------------------|
+            | 24-31: HeaderCollection <Headers>k__BackingField (8 bytes)      |
+            |-----------------------------------------------------------------|
+            | 32-39: Int64 _expectedBodyLength (8 bytes)                      |
+            |-----------------------------------------------------------------|
+            | 40-43: ResponseState _responseState (4 bytes)                   |
+            |-----------------------------------------------------------------|
+            | 44-47: AuthenticationSchemes _authChallenges (4 bytes)          |
+            |-----------------------------------------------------------------|
+            | 48-51: BoundaryType _boundaryType (4 bytes)                     |
+            |-----------------------------------------------------------------|
+            | 52-55: padding (4 bytes)                                        |
+            |-----------------------------------------------------------------|
+            | 56-71: Nullable`1 _cacheTtl (16 bytes)                          |
+            |-----------------------------------------------------------------|
+            | 72-639: HTTP_RESPONSE_V2 _nativeResponse (568 bytes)            |
+            |=================================================================|
+             */
+            _reasonPhrase = null;
             _nativeStream = null;
+            // RequestContext - kept
+            Headers.IsReadOnly = false;
+            Headers.Clear();
+            _expectedBodyLength = 0;
+            _responseState = ResponseState.Created;
+            _authChallenges = AuthenticationSchemes.None;
+            _boundaryType = BoundaryType.None;
             _cacheTtl = null;
-            _authChallenges = RequestContext.Server.Options.Authentication.Schemes;
+
+            _nativeResponse = default; // 568 bytes!
```

Comment:
Could either be smarter with clearing this 568 bytes or just clear the counts and not clear the headers

Reply:
Couldn't work out if not nulling the pointers in the KnownHeaders struct would cause an issue; so left as is.